src/imports/Components/1.3/OptionSelectorDelegate.qml

```diff
@@ -321,7 +321,7 @@ ListItem.Empty {
                 - parent.spacing * 2 - parent.anchors.leftMargin
 
             Toolkit.Label {
-                text: option.text === "" ? modelData : option.text
+                text: (option.text === "" && typeof modelData !== "undefined") ? modelData : option.text
```

Comment:
Why is `undefined` in quotes here?

Reply:
because 
```js
typeof modelData
```
is a string

Reply:
Why are you not just doing `modelData !== undefined` though, as pretty much all the other code checking against undefined, does?

Reply:
when just writing `modelData !== undefined` I get again a
```
ReferenceError: modelData is not defined
```

Reply:
I got the solution from here:
https://askubuntu.com/questions/527799/how-do-you-check-if-a-property-is-undefined-in-qml/552586#552586

Reply:
Can you please change it to just do `&& modelData !== undefined` instead? Thanks.

Reply:
I changed it although on my system, the `modelData !== undefined` still throws the same ReferenceError as the current code.

Reply:
Hmm. Is there a specific app or such which shows that? I don't see it occurring in the test output during the build. Perhaps we could add a test which shows the problem, so we can prevent it from happening again?

Reply:
This problem is showing up in all my self developed apps. Each time an OptionSelectorDelegate instance is being created. Interestingly, the above linked answer on askubuntu describes exactly what I find. Maybe it's a bug that's fixed with Qt 5.12?

Reply:
Not sure. I don't see a ReferenceError here with UI Toolkit Gallery app either. I still keep seeing the binding loop issue though, even with change here.

```
file:///usr/lib/arm-linux-gnueabihf/qt5/qml/Ubuntu/Components/1.3/OptionSelector.qml:330:13: QML ListView: Binding loop detected for property "itemHeight"
```

Reply:
I'm inclined to close this for now, as it's not clear what's going on, and I can't reproduce the given issue. Can you attach a minimal example which shows the issue happening, in the #105 ticket? We're trying to get the repo migrated to GitLab as we work on updating to 20.04 (and this one is lower in the dependency stack), and I'd like to reproduce the issue and try to understand why using `typeof` and comparing the string would work but checking against `undefined` would not before approving.

Thanks!